lib/model/autocomplete.dart

```diff
@@ -188,8 +188,71 @@ class MentionAutocompleteView extends ChangeNotifier {
     required Narrow narrow,
   }) {
     assert(narrow is! CombinedFeedNarrow);
+
+    final (streamId, topic) = switch (narrow) {
+      StreamNarrow() => (narrow.streamId, null),
+      TopicNarrow() => (narrow.streamId, narrow.topic),
+      // The CombinedFeedNarrow case should be impossible anyway.
+      DmNarrow() || CombinedFeedNarrow() => (null, null),
```

Comment:
Let's move the assert above about CombinedFeedNarrow to inside here ­— that way we're splitting up the cases for `narrow`'s type in just one place.

To make that comfortable, this will probably want to become a switch statement instead of expression.